### PR TITLE
Switch to store in messages and fix send icon

### PR DESCRIPTION
**Pull request by alshakerM**
Added store to messages.
Now you can send a message.

### CODE REVIEW

File: src/Messages/Messages.js [hunk 1]
```diff
@@ -137,7 +134,10 @@ export function Messages({ fromUserId }) {
               </svg>
             </div>
             <div className={styles.messagesSection}>
-              <div className={styles.messagesBody}>
+              <div
+                className={styles.messagesBody}
+                ref={(div) => div && (div.scrollTop = 100000)}
```
Comment (alshakero) [inline src/Messages/Messages.js:139]: Can you create an issue for this so we don't forget? 

"Clean up scrolling logic for messages container div".

Because it scrolls down whenever you re-render. To see the bug, scroll up, then type anything in the field, it will scroll down.

File: src/Messages/Messages.js [hunk 2]
```diff
@@ -194,21 +194,46 @@ export function Messages({ fromUserId }) {
...
-                    height="24"
-                    role="img"
-                    viewBox="0 0 48 48"
-                    width="24"
+                  <form
+                    className={styles.inputForm}
+                    onSubmit={(event) => {
+                      event.preventDefault();
+                      const message = event.target.messageBody.value;
+                      submitMessage(userIndex, message);
```
Comment (alshakero) [inline src/Messages/Messages.js:202]: `submitMessage` doesn't need `message`. It's already stored in the state. And as we saw in the post, it's more robust to use ids than indexes. Switch to user ID if you can.

File: src/Messages/Messages.js [hunk 3]
```diff
@@ -194,21 +194,46 @@ export function Messages({ fromUserId }) {
...
+                          height="24"
+                          role="img"
+                          viewBox="0 0 48 48"
+                          width="24"
+                        >
+                          <path d="M38.5 0h-29C4.3 0 0 4.3 0 9.5v29C0 43.7 4.3 48 9.5 48h29c5.2 0 9.5-4.3 9.5-9.5v-29C48 4.3 43.7 0 38.5 0zM45 38.5c0 3.6-2.9 6.5-6.5 6.5h-29c-3.3 0-6-2.5-6.4-5.6l8.3-8.3c.1-.1.3-.2.4-.2.1 0 .2 0 .4.2l6.3 6.3c1.4 1.4 3.6 1.4 5 0L35.9 25c.2-.2.6-.2.8 0l8.3 8.3v5.2zm0-9.4l-6.2-6.2c-1.3-1.3-3.7-1.3-5 0L21.3 35.3c-.1.1-.3.2-.4.2-.1 0-.2 0-.4-.2L14.2 29c-1.3-1.3-3.7-1.3-5 0L3 35.2V9.5C3 5.9 5.9 3 9.5 3h29C42.1 3 45 5.9 45 9.5v19.6zM11.8 8.2c-1.9 0-3.5 1.6-3.5 3.5s1.6 3.5 3.5 3.5 3.5-1.6 3.5-3.5-1.6-3.5-3.5-3.5z"></path>
+                        </svg>
+                      )}
+                    </button>
+                  </form>
```
Comment (alshakero) [inline src/Messages/Messages.js:236]: You can leave these changes in, not a big deal, but they're not part of the PR. If you want to keep them, please change the PR to be "Improve messages implementation" and list the improvements you made.

File: src/messagesStore/actions.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    type: 'SET_MESSAGE_FIELD_TEXT',
+    text,
+  };
+}
+export function submitMessage(userIndex, text) {
+  return {
+    type: 'SUBMIT_MESSAGE',
+    userIndex,
+    text,
+  };
```
Comment (alshakero) [inline src/messagesStore/actions.js:22]: `text` isn't used in the reducer. Can go.

```suggestion
export function submitMessage(userIndex) {
  return {
    type: 'SUBMIT_MESSAGE',
    userIndex,
  };
```

File: src/messagesStore/selectors.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+export function getMessages(state) {
+  return state.messages;
+}
+export function getIsLoading(state) {
+  return state.isLoading;
+}
```
Comment (alshakero) [inline src/messagesStore/selectors.js:6]: As we agreed, please only add selectors only as you need them. "The best code is no code at all".

**Review by alshakero — CHANGES_REQUESTED** [review]
Great work. Looks very nice. Left some comments.

File: src/Messages/Messages.js (hunk 1 above)
Comment (alshakerM) [reply src/Messages/Messages.js:139]: Created https://github.com/alshakerM/my-instagram/issues/53.

File: src/Messages/Messages.js (hunk 2 above)
Comment (alshakerM) [reply src/Messages/Messages.js:202]: Done.

File: src/Messages/Messages.js (hunk 3 above)
Comment (alshakerM) [reply src/Messages/Messages.js:236]: Okay
